apps/fyler/src/fyler_category_manager.erl

```diff
@@ -0,0 +1,672 @@
+    NextState ->
+      erlang:apply(?MODULE, NextState, [start, NewState#state{check_ref = undefined}])
+  end.
+
+tasks({check, Ref}, #state{check_ref = Ref} = State) ->
+  tasks(check, State);
+
+tasks({check, _}, State) ->
+  {next_state, tasks, State};
+
+tasks(Event, #state{} = State) ->
+  {next_state, NextState, NewState} = tasks_(Event, State),
+  case NextState of
+    tasks ->
+      {next_state, NextState, NewState};
+    NextState ->
+      erlang:apply(?MODULE, NextState, [start, NewState#state{check_ref = undefined}])
+  end.
+
+start_(start, #state{category = Category} = State) ->
+  Priorities = maps:get(priorities, ?Config(Category, #{}), #{}),
+  FreeNodes =
+    fun
+      (#pool{node = Node, enabled = true, category = C}, List) when C == Category -> [{Node, enabled} | List];
+      (#pool{node = Node, enabled = false, category = C}, List) when C == Category -> [{Node, disabled} | List];
+      (_, List) -> List
+    end,
+  Pools = ets:foldl(FreeNodes, [], ?T_POOLS),
+  case ets:info(Category) of
+    undefined ->
+      ets:new(Category, [public, named_table, {keypos, #current_task.id}, {heir, whereis(fyler_server), ok}]);
+    _ ->
+      ok
+  end,
+  NextState =
+    case is_enabled(Pools) of
+      true -> pools;
+      false -> zero
+    end,
+
+  InstanceList = maps:get(instances, ?Config(Category, #{}), []),
+  {ok, Re} = re:compile(?Regex),
+  IdToNode =
+    fun
+      (Id, {Map, Nodes}) ->
+        case re:run(aws_cli:instance(Id), Re, [{capture, [ip], list}]) of
+          {match, [Ip]} ->
+            Node = ?Node(Category, Ip),
+            {maps:put(Node, Id, Map), [Node | Nodes]};
+          nomatch ->
+            {Map, Nodes}
+        end
+    end,
+  {NodeToId, AllNodes} = lists:foldl(IdToNode, {#{}, []}, InstanceList),
+  PassiveNodes = lists:subtract(AllNodes, Pools),
+
+  {next_state, NextState, State#state{pools = Pools, passive_pools = PassiveNodes, priorities = Priorities, node_to_id = NodeToId}}.
+
+zero_(start, State) ->
+  {next_state, zero, State};
+
+zero_({run, #task{priority = Priority} = Task}, #state{tasks = Tasks, priorities = Priorities} = State) ->
+  NumPriority = maps:get(Priority, Priorities, 1),
+  update_ets(Task),
+  {next_state, tasks, State#state{tasks = fyler_queue:in(Task, NumPriority, Tasks)}};
+
+zero_({task_rejected, Ref, Node}, #state{category = Category, retasks = ReTasks, pools = Pools} = State) ->
+  case status(Node, Pools) of
+    {pending, Ref, Id} ->
+      NewPools = enable(Node, Pools),
+      case task(Category, Id) of
+        undefined ->
+          {next_state, zero, State#state{pools = NewPools}};
+        Task ->
+          {NextState, NewState} = choose_pool(State#state{retasks = fyler_queue:in(Task, ReTasks), pools = NewPools}),
+          {next_state, NextState, NewState}
+      end;
+    _ ->
+      {next_state, zero, State}
+  end;
+
+zero_({pool_connected, Node}, #state{pools = Pools, passive_pools = PassivePools} = State) ->
```

Comment:
Line 200 is too long: zero_({pool_connected, Node}, #state{pools = Pools, passive_pools = PassivePools} = State) ->.